pandas/tests/io/test_sql.py

```diff
@@ -817,6 +817,48 @@ def psql_insert_copy(table, conn, keys, data_iter):
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.db
+@pytest.mark.parametrize("conn", postgresql_connectable)
+def test_self_join_date_columns(conn, request):
+    # GH 44421
+    conn = request.getfixturevalue(conn)
```

Comment:
Can we add this to the Postgres class? The classes are skipped when no db is detected

Reply:
Sure moved (though I think eventually we should move these classes to the functional fixture instead of the class based ones)